File: 2014-10-CSharp-Fundamentals-1/8. Multidimensional Arrays/BinarySearch/BinarySearch.cs
using System;
using System.Collections.Generic;
using System.Linq;

class BinarySearch
{
    static Random rand = new Random();

    static void Main()
    {
        string[] beers =
        {
            "Zagorka", "Ariana", "Shumensko", "Astika",
            "Kamenitza", "Bolyarka", "Amstel"
        };
        Array.Sort(beers);

        string target = "Astika";
        int index = Array.BinarySearch(beers, target);
        Console.WriteLine("{0} is found at index {1}.", target, index);
        // Result: Astika is found at index 2.

        target = "Heineken";
        index = Array.BinarySearch(beers, target);
        Console.WriteLine("{0} is not found (index={1}).", target, index);
        // Result: Heineken is not found (index=-5).

        //load test

        string line = Console.ReadLine();
        int n = int.Parse(line);
        byte[] array = new byte[n];

        for (int i = 0; i < n; i++)
        {
            array[i] = (byte)(i % byte.MaxValue);
        }

        Array.Sort(array);

        DateTime start = DateTime.Now;
        foreach (var element in array)
        {
            Array.IndexOf(array, element);
        }
        DateTime end = DateTime.Now;

        Console.WriteLine("IndexOf: {0}", end - start);

        start = DateTime.Now;
        foreach (var element in array)
        {
            Array.BinarySearch(array, element);
        }
        end = DateTime.Now;

        Console.WriteLine("BinarySearch: {0}", end - start);
    }
}

File: 2015-10-Java-Fundamentals-1/07. Arrays/demos/BinarySearch.java
import java.text.MessageFormat;
import java.util.Arrays;

public class BinarySearch {
    public static void main(String[] args) {
        String[] beers = { "Zagorka", "Ariana", "Shumensko",
                "Astika", "Kamenitza", "Bolyarka", "Amstel"
        };
        Arrays.sort(beers);

        String target = "Astika";
        int index = Arrays.binarySearch(beers, target);
        writeLine("{0} is found at index {1}.", target, index);
        //// Result: Astika is found at index 2.

        target = "Heineken";
        index = Arrays.binarySearch(beers, target);
        writeLine("{0} is not found (index={1}).", target, index);
        //// Result: Heineken is not found (index=-5).
    }

    public static void writeLine(String format, Object... obj) {
        MessageFormat form = new MessageFormat(format);
        System.out.println(form.format(obj));
    }
}